File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.cs
using System;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int num, int total) {
        List<int> answer = new List<int>();
        float comp;
        
        if(num % 2 != 0)
        {
            comp = total/num;
            answer.Add((int)comp);
            for(int i = 1; i <= num / 2; i++)
            {
                answer.Add((int)comp - i);
                answer.Add((int)comp + i);
            }
        }
        else
        {
            comp = (float)total/num;
            for(int i = 1; i <= num / 2; i++)
            {
                answer.Add((int)Math.Ceiling(comp - i));
                answer.Add((int)Math.Floor(comp + i));
            }
        }
        answer.Sort();
        return answer.ToArray();
    }
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int num, int total) {
    vector<int> answer;
    int comp = total/num;
    for(int i = 0; i < num; i++)
    {
        int data = comp - (num/2) + i;
        if(num % 2 == 0)
            data++;
        answer.push_back(data);
    }
    return answer;
}